platforms/posix/cmake/sitl_target.cmake

```diff
@@ -57,94 +58,161 @@ ExternalProject_Add(mavsdk_tests
-set(models none shell
-	if750a iris iris_opt_flow iris_opt_flow_mockup iris_vision iris_rplidar iris_irlock iris_obs_avoid iris_rtps solo typhoon_h480
-	plane plane_cam plane_catapult
-	standard_vtol tailsitter tiltrotor
-	rover boat
-	uuv_hippocampus)
-set(worlds none empty baylands ksql_airport mcmillan_airfield sonoma_raceway warehouse)
-set(all_posix_vmd_make_targets)
-foreach(viewer ${viewers})
-	foreach(debugger ${debuggers})
-		foreach(model ${models})
-			foreach(world ${worlds})
-				if (world STREQUAL "none")
-					if (debugger STREQUAL "none")
-						if (model STREQUAL "none")
-							set(_targ_name "${viewer}")
-						else()
-							set(_targ_name "${viewer}_${model}")
-						endif()
-					else()
-						if (model STREQUAL "none")
-							set(_targ_name "${viewer}___${debugger}")
-						else()
-							set(_targ_name "${viewer}_${model}_${debugger}")
-						endif()
-					endif()
-
-					add_custom_target(${_targ_name}
-					COMMAND ${PX4_SOURCE_DIR}/Tools/sitl_run.sh
-						$<TARGET_FILE:px4>
-						${debugger}
-						${viewer}
-						${model}
-						${world}
-						${PX4_SOURCE_DIR}
-						${PX4_BINARY_DIR}
-					WORKING_DIRECTORY ${SITL_WORKING_DIR}
-					USES_TERMINAL
-					DEPENDS
-						logs_symlink
-					)
-					list(APPEND all_posix_vmd_make_targets ${_targ_name})
-					if (viewer STREQUAL "gazebo")
-						add_dependencies(${_targ_name} px4 sitl_gazebo)
-					elseif(viewer STREQUAL "jmavsim")
-						add_dependencies(${_targ_name} px4 git_jmavsim)
-					endif()
-				else()
-					if (viewer STREQUAL "gazebo")
-						if (debugger STREQUAL "none")
-							if (model STREQUAL "none")
-								set(_targ_name "${viewer}___${world}")
-							else()
-								set(_targ_name "${viewer}_${model}__${world}")
-							endif()
-						else()
-							if (model STREQUAL "none")
-								set(_targ_name "${viewer}__${debugger}_${world}")
-							else()
-								set(_targ_name "${viewer}_${model}_${debugger}_${world}")
-							endif()
-						endif()
-
-						add_custom_target(${_targ_name}
-						COMMAND ${PX4_SOURCE_DIR}/Tools/sitl_run.sh
-							$<TARGET_FILE:px4>
-							${debugger}
-							${viewer}
-							${model}
-							${world}
-							${PX4_SOURCE_DIR}
-							${PX4_BINARY_DIR}
-						WORKING_DIRECTORY ${SITL_WORKING_DIR}
-						USES_TERMINAL
-						DEPENDS
-							logs_symlink
-						)
-						list(APPEND all_posix_vmd_make_targets ${_targ_name})
-						add_dependencies(${_targ_name} px4 sitl_gazebo)
-					endif()
-				endif()
-			endforeach()
-		endforeach()
-	endforeach()
-endforeach()
+ExternalProject_Add(flightgear_bridge
+	SOURCE_DIR ${PX4_SOURCE_DIR}/Tools/flightgear_bridge
+	CMAKE_ARGS
+		-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+	BINARY_DIR ${PX4_BINARY_DIR}/build_flightgear_bridge
+	INSTALL_COMMAND ""
+	DEPENDS
+		git_flightgear_bridge
+	USES_TERMINAL_CONFIGURE true
+	USES_TERMINAL_BUILD true
+	EXCLUDE_FROM_ALL true
+	BUILD_ALWAYS 1
+)
+
+
+if( ENABLE_LOCKSTEP_SCHEDULER STREQUAL "yes")  
```

Comment:
Can you elaborate on this condition? why do we need it? we still need to be able to run jmavsim and gazebo without lockstep

Reply:
Thank you - I will clear it.

Reply:
Done. sitl_target.cmake  simlified and gazebo and etc added back for build without lockstep.

Reply:
Thanks! Looks much better now